.circleci/config.yml

```diff
@@ -468,49 +464,56 @@ jobs:
+            GITHUB_PROJECT="$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
+            LIST_RESP=$(request "https://api.github.com/repos/$GITHUB_PROJECT/actions/runs/$GH_ACTIONS_RUN_ID/artifacts")
             if [[ "${LIST_RESP:0:5}" == "Error" ]]; then
               echo "List response seems to be an error"
               echo "$LIST_RESP"
               exit 1
             fi
-            FILE_LIST_URL=$(echo "$LIST_RESP" | jq -rc '.value[0].fileContainerResourceUrl //empty')
-            LIST_JSON_FILE=$(request GET "$FILE_LIST_URL" 1)
-            while read FILE_PATH; do
-              DOWNLOAD_URL=$(cat $LIST_JSON_FILE | jq -rc ".value[] | select(.path==\"$FILE_PATH\") | .contentLocation")
-              FILE_SIZE=$(cat $LIST_JSON_FILE | jq -rc ".value[] | select(.path==\"$FILE_PATH\") | .fileLength")
 
-              FILE_PATH=${FILE_PATH#*/}
-              echo "Downloading $FILE_PATH (size: $FILE_SIZE)"
-              mkdir -p $(dirname $BASE_DIR/$FILE_PATH)
-              FILE_SAVED=$(request GET "$DOWNLOAD_URL" 1)
-              mv $FILE_SAVED $BASE_DIR/$FILE_PATH
-            done < <(cat $LIST_JSON_FILE | jq -rc ".value[] | select((.itemType==\"file\") and (.path|startswith(\"$GH_ACTIONS_BUILD_ARTIFACT_NAME/\"))) | .path")
+            TARGET_ARTIFACT=$(echo "$LIST_RESP" | tr '\n' ' ' | jq -Rrc "fromjson | .artifacts[] | select(.name==\"$GH_ACTIONS_BUILD_ARTIFACT_NAME\") //empty")
+            if [[ "$TARGET_ARTIFACT" == "" ]]; then
+              echo "Could not find artifact $GH_ACTIONS_BUILD_ARTIFACT_NAME in run $GH_ACTIONS_RUN_ID."
+              exit 1
+            fi
+            if [[ "$(echo $TARGET_ARTIFACT | jq -rc '.expired')" == "true" ]]; then
+              echo "Artifact $GH_ACTIONS_BUILD_ARTIFACT_NAME in run $GH_ACTIONS_RUN_ID has expired."
+              exit 1
+            fi
+
+            DOWNLOAD_URL=$(echo $TARGET_ARTIFACT | jq -rc ".archive_download_url")
+            FILE_SIZE=$(echo $TARGET_ARTIFACT | jq -rc ".size_in_bytes")
 
+            echo "Downloading... (size: $FILE_SIZE)"
+            FILE_SAVED=$(request "$DOWNLOAD_URL" 1)
+            mv $FILE_SAVED ${FILE_SAVED}.zip
             echo "Downloading complete."
 
             echo "Extracting files..."
-            ARCH='<< parameters.arch >>'
-            ARTIFACT=$(find $BASE_DIR/distributions/out/*-linux-$ARCH.tar.gz)
-            if [[ "$ARTIFACT" == "" ]]; then
+            unzip ${FILE_SAVED}.zip -d $BASE_DIR
+
+            TARBALL=$(find $BASE_DIR/distributions/out/*-linux-$ARCH.tar.gz)
+            if [[ "$TARBALL" == "" ]]; then
               echo "Could not find built artifact for linux($ARCH)."
               exit 1
             fi
-            cat $ARTIFACT.sha256 | sha256sum --check
+            cat ${TARBALL}.sha256 | sha256sum --check
 
             mkdir $BASE_DIR/distributions/artifacts
-            tar -xzf $ARTIFACT -C $BASE_DIR/distributions/artifacts
+            tar -xzf $TARBALL -C $BASE_DIR/distributions/artifacts
```

Comment:
If this uses public APIs now can we just use `gh` and not have to deal with all the bash code?

Reply:
Sure, good idea! 
Just had a look at the `gh` CLI, it worked like a charm. So I've just updated the code.

Reply:
We'll still need this block throug. What it does is restoring the folder structure after the downloading, so it's required regardless of how we download.